src/backend/postmaster/seccomp.c

```diff
@@ -113,37 +110,51 @@ void seccomp_load_rules(PgSeccompRule *rules, int count)
 	 */
 	action.sa_sigaction = seccomp_test_sighandler;
 	if (sigaction(SIGSYS, &action, NULL) != 0)
-		raise_error("failed to install a test SIGSYS handler");
+		ereport(FATAL,
+				(errcode(ERRCODE_SYSTEM_ERROR),
+				 errmsg("seccomp: could not install test SIGSYS handler")));
 
 	/*
 	 * First, check that open of a well-known file works.
 	 * XXX: We use raw syscall() to call the very open().
 	 */
 	fd = syscall(SCMP_SYS(open), "/dev/null", O_RDONLY, 0);
-	if (fd < 0 || seccomp_test_sighandler_done)
-		raise_error("failed to open a test file");
-	close((int)fd);
+	if (seccomp_test_sighandler_done)
+		ereport(FATAL,
+				(errcode(ERRCODE_SYSTEM_ERROR),
+				 errmsg("seccomp: signal handler test flag was set unexpectedly")));
+	if (fd < 0)
+		ereport(FATAL,
+				(errcode(ERRCODE_SYSTEM_ERROR),
+				 errmsg("seccomp: could not open /dev/null for seccomp testing: %m")));
+	close((int) fd);
 
 	/* Set a trap on open() to test seccomp bpf */
 	rule = PG_SCMP(open, SCMP_ACT_TRAP);
 	if (do_seccomp_load_rules(&rule, 1, SCMP_ACT_ALLOW) != 0)
-		raise_error("failed to load a test filter");
+		ereport(FATAL,
+				(errcode(ERRCODE_SYSTEM_ERROR),
+				 errmsg("seccomp: could not load test trap")));
```

Comment:
NIT: Looks like the error raising has quite a pattern in all changed places, I wonder if we could generalize it?